extensions/Worker.Extensions.Http.AspNetCore/src/FromBodyConverstionFeature.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.Azure.Functions.Worker.Extensions.Http.Converters;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore
+{
+    internal sealed class FromBodyConverstionFeature : IFromBodyConversionFeature
+    {
+        public static FromBodyConverstionFeature Instance { get; } = new();
+
+        public async ValueTask<object?> ConvertAsync(FunctionContext context, Type targetType)
+        {
+            var httpContext = context.GetHttpContext()
+                ?? throw new InvalidOperationException($"The '{nameof(FromBodyConverstionFeature)} expects an '{nameof(HttpContext)}' instance in the current context.");
+
+            var metadata = httpContext.RequestServices
+                           .GetService<IModelMetadataProvider>()?
+                           .GetMetadataForType(targetType);
+
+            if (metadata is not null)
+            {
+                var binderModelName = metadata.BinderModelName ?? string.Empty;
+
+                var modelBinder = httpContext.RequestServices
+                   .GetService<IModelBinderFactory>()?
+                   .CreateBinder(new ModelBinderFactoryContext
+                   {
+                       Metadata = metadata,
+                       BindingInfo = new BindingInfo()
+                       {
+                           BinderModelName = binderModelName,
+                           BindingSource = BindingSource.Body,
+                           BinderType = metadata.BinderType,
+                           PropertyFilterProvider = metadata.PropertyFilterProvider,
+                       },
+                       CacheToken = null
+                   });
+
+                if (modelBinder is not null)
+                {
+                    var modelBindingContext = new DefaultModelBindingContext
+                    {
+                        ModelMetadata = metadata,
+                        ModelName = binderModelName,
+                        BindingSource = BindingSource.Body,
+                        ModelState = new ModelStateDictionary(),
+                        ActionContext = new ActionContext
+                        {
+                            HttpContext = httpContext
+                        }
+                    };
+
+                    await modelBinder.BindModelAsync(modelBindingContext);
+
+                    if (modelBindingContext.Result.IsModelSet)
+                    {
+                        return modelBindingContext.Result.Model;
+                    }
+                    else if (!modelBindingContext.ModelState.IsValid)
```

Comment:
what scenario would trigger this? the payload can't be bound to the object?

Reply:
Yes. A request without a body payload that can be deserialized into the target type or a validation failure are some examples.